platform-connectors/pkg/nodemetadata/processor.go

```diff
@@ -0,0 +1,153 @@
+// Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package nodemetadata
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+
+	pb "github.com/nvidia/nvsentinel/data-models/pkg/protos"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+// processor implements the Processor interface using LRU cache with coordinated fetching.
+type processor struct {
+	config    *Config
+	clientset kubernetes.Interface
+	cache     *Cache
+	stopCh    chan struct{}
+}
+
+// NewProcessor creates a new node metadata processor.
+func NewProcessor(ctx context.Context, config *Config) (Processor, error) {
+	if err := config.Validate(); err != nil {
+		return nil, fmt.Errorf("invalid config: %w", err)
+	}
+
+	restConfig, err := rest.InClusterConfig()
```

Comment:
We need to build from flags using clientcmd. I suspect we already have another client that is already built with the correct QPS etc, let's use the same.